staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -474,13 +511,13 @@ func determineResyncPeriod(desired, check time.Duration) time.Duration {
 
 const minimumResyncPeriod = 1 * time.Second
 
-func (s *sharedIndexInformer) AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration) {
+func (s *sharedIndexInformer) AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration) (ResourceEventHandlerRegistration, error) {
 	s.startedLock.Lock()
 	defer s.startedLock.Unlock()
 
 	if s.stopped {
-		klog.V(2).Infof("Handler %v was not added to shared informer because it has stopped already", handler)
-		return
+		klog.V(2).Infof("Handler %v is not added to shared informer because it has stopped already", handler)
```

Comment:
I think we should remove the log line now that we're returning an error. It's a little rude for libraries to log internally anyway, it limits choices of the consumer.